python_modules/dagster/dagster/_core/definitions/multi_asset_sensor_definition.py

```diff
@@ -851,9 +845,8 @@ def get_asset_cursor_with_advances(
                 initial_asset_cursor.trailing_unconsumed_partitioned_event_ids
             )
             unconsumed_events = list(context.get_trailing_unconsumed_events(asset_key)) + list(
-                context.instance.get_event_records(
-                    EventRecordsFilter(
-                        event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                context.instance.get_materialization_records(
+                    AssetRecordsFilter(
```

Comment:
i assume this would be fine if it used a limit on the fetch ? just would update its cursors accordingly and progress forward next tick?